src/CoolPropLib.cpp

```diff
@@ -419,6 +419,24 @@ EXPORT_CODE long CONVENTION AbstractState_factory(const char* backend, const cha
 	}
     return -1;
 }
+EXPORT_CODE void CONVENTION AbstractState_fluid_names(const long handle, char **& fluids, long *N, long *errcode, char *message_buffer, const long buffer_length)
+{
+    *errcode = 0;
+
+    try{
+        shared_ptr<CoolProp::AbstractState> &AS = handle_manager.get(handle);
+        std::vector<std::string> _fluids = AS->fluid_names();
+        *N = _fluids.size();
+        fluids = (char**)malloc(sizeof(char*)**N);                        // fluids is only allocated memory if AbstractState exists for handle --> only needs to be freed, if *N != 0
```

Comment:
Memory leak. No low-level memory allocation, please use the string joining functions in CPStrings.h